meta-resin-common/recipes-connectivity/networkmanager/networkmanager/balena-client-id.patch

```diff
@@ -0,0 +1,39 @@
+From dfe53c829e24f1240a2ac8ef8819b506c7d22cfc Mon Sep 17 00:00:00 2001
+From: Sebastian Panceac <sebastian@resin.io>
+Date: Wed, 17 Oct 2018 14:36:56 +0200
+Subject: Replace NM's DHCP request option "Client indentifier" with udhcpc style option
+
+Signed-off-by: Sebastian Panceac <sebastian@balena.io>
+
+Upstream-Status: Pending
+---
+ src/systemd/src/libsystemd-network/sd-dhcp-client.c | 8 ++++++--
+ 1 file changed, 6 insertions(+), 2 deletions(-)
+
+diff --git a/src/systemd/src/libsystemd-network/sd-dhcp-client.c b/src/systemd/src/libsystemd-network/sd-dhcp-client.c
+index c2f81e1..16962db 100644
+--- a/src/systemd/src/libsystemd-network/sd-dhcp-client.c
++++ b/src/systemd/src/libsystemd-network/sd-dhcp-client.c
+@@ -627,13 +627,17 @@ static int client_message_init(
+                 client->client_id_len = sizeof(client->client_id.type) + sizeof(client->client_id.ns.iaid) + duid_len;
+         }
+ 
++        uint8_t balena_client_id[7] = {0};
++        balena_client_id[0] = 1; /* Hardware type: Ethernet */
```

Comment:
@spanceac is it ok to set this byte to Ethernet in case of wifi connection?

Reply:
@roman-mazur, that's how udhcpc did it regardless of the transfer media. We do the same here and it didn't prove to be a problem

Reply:
Thanks for the info.
I just wonder why it's not just the MAC address... but something with a magic prepended byte...